app-backend/db/src/main/scala/com/azavea/rf/database/ObjectPermissions.scala

```diff
@@ -166,7 +144,7 @@ trait ObjectPermissions[Model] {
 
   def deletePermissions(
       id: UUID): ConnectionIO[List[Option[ObjectAccessControlRule]]] =
-    updatePermissionsF(id, List[ObjectAccessControlRule]()).update
+    updatePermissionsF(id, List[ObjectAccessControlRule](), replace = true).update
```

Comment:
I am not sure if passing `replace = true` is necessary, since we have already passed an empty list of ACRs to `updatePermissionsF `, which should set `acr = '{}'::text[]` just fine as reflected in the first case of `acrList.length`.

But passing this parameter in does not seem to hurt anything.

Reply:
Ah woops, you're right, I didn't notice that this would get covered in the match on `acrList.length` in `updatePermissionsF`. I'll change it

Reply:
I also made a change to pattern match on `aList` or `aString` directly instead of `aList.length` or `aString.length`, because:

>  Time: List has O(1) prepend and head/tail access. Most other operations are O(n) on the number of elements in the list. This includes the index-based lookup of elements, length, append and reverse.

Probably it'll never matter in the context of ACRs or annotation label length, but it's a good pattern to have in the backend code.

https://www.scala-lang.org/api/2.11.12/#scala.collection.immutable.List